src/ensembl/src/content/app/species/components/species-stats/SpeciesStats.tsx

```diff
@@ -0,0 +1,102 @@
+/**
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import classNames from 'classnames';
+
+import defaultStyles from './SpeciesStats.scss';
+
+export type SpeciesStatsProps = {
+  title?: string;
+  label: string;
+  labelHint?: string;
+  primaryValue: string;
+  primaryUnit?: string;
+  secondaryValue?: string;
+  secondaryUnit?: string;
+  link?: React.ReactElement;
+  classNames?: {
+    wrapper?: string;
+    title?: string;
+    label?: string;
+    labelHint?: string;
+    primaryValue?: string;
+    primaryUnit?: string;
+    secondaryValue?: string;
+    secondaryUnit?: string;
+    link?: string;
+  };
+};
```

Comment:
I have an alternative proposal. What if instead of passing all the props to a single component and making it figure it all out, we split the component into several smaller components.

For example, here are components with three types of labels (the second type Andrea is likely to retire, but notice, too, how weird it is, with the label growing upwards instead of pushing the main content down):

![image](https://user-images.githubusercontent.com/6834224/91506462-07325b80-e8ca-11ea-8b9c-b1becd72b613.png)

```jsx
// focusing just on the label

// left example
<SpeciesStats>
  <SpeciesStatsLabel>
    Coding genes
  </SpeciesStatsLabel>
</SpeciesStats>

// middle example
<SpeciesStats>
  <SpeciesStatsLabel2
    line1="No. genes in"
    line2="Gene families"
  />
</SpeciesStats>

// or possibly:
<SpeciesStats>
  <SpeciesStatsLabel>
    <span className="light">No. genes in</span><br/>
    <span className="bold">Gene families</span><br/>
  </SpeciesStatsLabel>
</SpeciesStats>

// right example
<SpeciesStats>
  <SpeciesStatsLabel>
    SNVs <span className="small">single nucleotide variants<span>
  </SpeciesStatsLabel>
</SpeciesStats>
```

The main area:

![image](https://user-images.githubusercontent.com/6834224/91506699-a35c6280-e8ca-11ea-8ef0-343c4383342c.png)

```jsx

// left
<SpeciesStats>
  <SpeciesStatsLabel>
    Total sequence length
  </SpeciesStatsLabel>
  <SpeciesStatsMain>
    4,537,931,177 <span className="small">bp</span>
  </SpeciesStatsLabel>
</SpeciesStats>

// middle
<SpeciesStats>
  <SpeciesStatsLabel>
    Repeat regions
  </SpeciesStatsLabel>
  <SpeciesStatsMain>
    30%
  </SpeciesStatsLabel>
</SpeciesStats>

// right
<SpeciesStats>
  <SpeciesStatsMain>
    23 <span className="small">chromosomes or plasmids</span>
  </SpeciesStatsLabel>
</SpeciesStats>
```

and all together now:

![image](https://user-images.githubusercontent.com/6834224/91506749-c5ee7b80-e8ca-11ea-81f7-84d565957d9d.png)

```jsx

// left
<SpeciesStats>
  <SpeciesStatsLabel>
    Coding genes
  </SpeciesStatsLabel>
  <SpeciesStatsMain>
    20,438
  </SpeciesStatsLabel>
  <SpeciesStatsSecondary>
    671 <span className="small">readthrough</span>
  </SpeciesStatsSecondary>
</SpeciesStats>

// right
<SpeciesStats>
  <SpeciesStatsLabel>
    Homology data
  </SpeciesStatsLabel>
  <SpeciesStatsMain>
    96%
  </SpeciesStatsLabel>
  <SpeciesStatsSecondary>
    <Link to="/some/path">view</Link>
  </SpeciesStatsSecondary>
</SpeciesStats>
```

I think this offers a higher flexibility and less confusion that a single component with numerous optional props might cause.

Reply:
The main reason that convinced me to create a single component that all the stats share a similar structure. I am sure how breaking it down to smaller components like in the example above is going to be different from the original plan of just having the divs and classnames. 

Also, with the above example we have to import all the components and then manually build each stat based on it's structure.

Reply:
> all these stats share a similar structure

But notice how many nulls/undefineds/optional_fields you have in the props type for this component. Notice also how many different props there are. It suggests that the components may not have such a similar structure after all.

There are several talks addressing these issues. One (although very imperfect, with metaphors that do not quite work), is [React is Fiction](https://www.youtube.com/watch?v=kqh4lz2Lkzs); another, much better, is [AHA programming](https://youtu.be/wuVy7rwkCfc), by Kent C. Dodds. The general feeling of the community is that there's something not quite right with this approach:

![image](https://user-images.githubusercontent.com/6834224/91839486-080b1a80-ec47-11ea-84f1-eade36b27ac4.png)

> I am not sure how breaking it down to smaller components like in the example above is going to be different from the original plan of just having the divs and classnames.

It will capture common styling in the stats component and its subcomponents. Compared to the option of just using classnames of the parent component, it offers some degree of encapsulation of the styles.

Reply:
Should we really consider 9 props as too many? We already have input and textarea component that takes more than 9 pros. Almost all the components that are available in Ant design has several optional props.

If we go with your suggestion, how do we represent an array of individual json stats without having to hardcode the structure for each stat?

Reply:
Decisions after code review:

- go ahead with the component structure proposed in the PR
- rename some of the props:
  - `title` -> `preLabel`
  - `labelHint` - remove

Reply:
That's done